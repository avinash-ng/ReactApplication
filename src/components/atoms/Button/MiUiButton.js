import React from 'react'
import { Button } from "@material-ui/core"
const MiUiButton = props => {
    return (
        <Button className={props.className} style={props.style} variant={props.variant} onClick={props.onClick}>{props.buttonName}</Button> 
    )
}
export default MiUiButton