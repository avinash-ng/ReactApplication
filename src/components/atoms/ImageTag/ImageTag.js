import React from 'react'
import './ImageTag.css'
const ImageTag = props => {
    return (
        <h3 className={props.className} style={{ width: '450px', marginBottom: '20px' }} onClick={props.onClicked} >{props.object.name}</h3>
    )
}
export default ImageTag