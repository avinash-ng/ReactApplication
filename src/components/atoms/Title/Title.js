import React from 'react'
import { Typography } from '@material-ui/core'
import { useAuth0 } from '../../../react-auth0-spa'

const Title = (props) => {
    
    return (
        // <h6> Hi {props.title} Welcome</h6>
        <Typography variant="h6" noWrap>
                Hi {props.name}      
        </Typography>
    )
}

export default Title;