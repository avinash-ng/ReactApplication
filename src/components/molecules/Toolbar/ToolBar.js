import { Toolbar } from "@material-ui/core"
import Title from "../../atoms/Title/Title"
import MiUiButton from '../../atoms/Button/MiUiButton'
import React from 'react'
import {makeStyles} from '@material-ui/core/styles'
// import './ToolBar.css'
const useStyles = makeStyles((theme) => ({
    button: {
        marginLeft: '1000px'  
    }

})) 
const ToolBar = (props) => {
    const classes = useStyles()
    return (
        <Toolbar>
            <Title name={props.name}/>
            <MiUiButton className={classes.button} style={props.style} className="button" onClick={props.onClick} buttonName={props.buttonName}> </MiUiButton>
        </Toolbar>
    )
}

export default ToolBar