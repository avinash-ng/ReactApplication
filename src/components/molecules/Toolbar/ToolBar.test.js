import ToolBar from './ToolBar'
import expect from 'expect'
import sinon from 'sinon';
import { shallow, configure, mount } from 'enzyme'
import React from 'react'
import Adapter from 'enzyme-adapter-react-16';
import { Button } from '@material-ui/core';
import ImageTag from '../../atoms/ImageTag/ImageTag'
import MiUiButton from '../../atoms/Button/MiUiButton'
import Title from '../../atoms/Title/Title';

configure({ adapter: new Adapter() });
describe('<ToolBar /> ', () => {
    it('renders Title /> element properly', () => {
        
        const RenderComponent = shallow(
            <ToolBar ></ToolBar>
        );
        // console.log(RenderComponent)
        expect(RenderComponent.find(Title).exists()).toBe(true)
    })
    
    it('renders <MiUiButton /> element properly', () => {
        
        const buttonName='click me'
        const RenderComponent = shallow(
            <ToolBar ></ToolBar>
        );
        // console.log(RenderComponent)
        expect(RenderComponent.find(MiUiButton).exists()).toBe(true)
    })
    it('recieved props correctly and check on click functionality', () => {
        const name = "Welcome"
        const onClick = sinon.spy()
        const  buttonName='click me'
        const RenderComponent = shallow(<ToolBar name={name} onClick={onClick} buttonName={buttonName}></ToolBar>)
        RenderComponent.find(MiUiButton).simulate('click')
        expect(onClick).toHaveProperty('callCount', 1)
    })


    it('Matches snapshot', () => {
        const RenderComponent = shallow(<ToolBar></ToolBar>)
        expect(RenderComponent).toMatchSnapshot()
    })
})