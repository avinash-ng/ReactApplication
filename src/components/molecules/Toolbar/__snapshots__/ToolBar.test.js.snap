// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ToolBar />  Matches snapshot 1`] = `ShallowWrapper {}`;
